pyee/util.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+
+from functools import wraps
+
+
+def _wrap(left, right, error_handler, proxy_new_listener):
+    left_emit = left.emit
+    right_emit = right.emit
+
+    @wraps(left_emit)
+    def wrapped_emit(event, *args, **kwargs):
+        left_handled = left._call_handlers(event, args, kwargs)
+
+        # Do it for the right side
+        if proxy_new_listener or event != 'new_listener':
+            right_handled = right._call_handlers(event, args, kwargs)
+        else:
+            right_handled = False
+
+        handled = left_handled or right_handled
+
+        # Use the error handling on ``error_handler`` (should either be
+        # ``left`` or ``right``)
+        if not handled:
+            error_handler._emit_handle_potential_error(
+                event, args[0] if args else None
+            )
+
+        return handled
+
+    def unwrap():
```

Comment:
we could grab the original unwraps too, so that double wrapping at least own't lead to the inability to fully unwrap